### PR TITLE
create internal solution for usaco-1180 (non-transitive dice)

### PR DESCRIPTION
**Pull request by NimeeshS**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/bronze/usaco-1180.mdx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+## Explanation
+
+To begin with this problem, we must first understand what it means for three dice to be transitive. Dice are transitive when dice $A$ beats dice $B$, dice $B$ beats dice $C$, and dice $C$ beats dice $A$. Dice $A$ beats dice $B$ when there are more pairs $(x, y)$ where $x$ is a side of dice $A$ and $y$ is a side on dice $B$ where $x > y$ compared to when $y < x$.
+
+Since each dice has four sides and there are ten possible values for each side, there are $10^4 = 10,000$ possible dice to test. This number is small enough that we can test every possible dice for non-transitivity via brute force.
+
+## Implementation
+
+**Time Complexity**: $\mathcal{O}(t \cdot 10^4)$
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:18]: constant factors aren’t part of the time comp

File: solutions/bronze/usaco-1180.mdx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	static boolean beats(int[] dice1, int[] dice2) {
+        int diff = 0;
+        for (int x : dice1) {
+            for (int y : dice2) {
+                diff += Integer.signum(x - y);
+            }
+        }
+        return diff > 0;
+    }
+
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:79]: weird indentation

File: solutions/bronze/usaco-1180.mdx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            for (int j = 0; j < 4; j++) {
+                diceA[j] = Integer.parseInt(tokenizer.nextToken());
+            }
+            int[] diceB = new int[4];
+            for (int j = 0; j < 4; j++) {
+                diceB[j] = Integer.parseInt(tokenizer.nextToken());
+            }
+            String ans = "no";
+            // Iterate through all 10,000 possibilities and check if there is a possible transitive dice
+			for (int w = 1; w <= 10; w++) {
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:95]: same here

File: solutions/bronze/usaco-1180.mdx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                for (int x = 1; x <= 10; x++) {
+                    for (int y = 1; y <= 10; y++) {
+                        for (int z = 1; z <= 10; z++) {
+                            int[] diceC = {w, x, y, z};
+                            if (beats(diceA, diceB) && beats(diceB, diceC) && beats(diceC, diceA)) ans = "yes";
+                            if (beats(diceB, diceA) && beats(diceA, diceC) && beats(diceC, diceB)) ans = "yes";
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:101]: braces

File: solutions/bronze/usaco-1180.mdx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+```py
+# Create method to calculate whether dice A beats dice B
+def win(a, b):
+    return sum([x > y for x in a for y in b]) > sum([y > x for x in a for y in b])
+
+t = int(input())
+for _ in range(t):
+    l = [int(x) for x in input().split()]
+    a_die = l[0:4]
+    b_die = l[4:8]
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:126]: use tabs to indent

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/bronze/usaco-1180.mdx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+using Die = array<int, 4>;
+
+/* Create method to calculate whether dice A beats dice B */
+bool beats(const Die &a, const Die &b) {
+	int wins = 0, losses = 0;
+	for (int i = 0; i < 4; i++)
+		for (int j = 0; j < 4; j++) {
+			if (a[i] > b[j]) ++wins;
+			if (a[i] < b[j]) ++losses;
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:35]:
```suggestion
			if (a[i] > b[j]) { wins++; }
			if (a[i] < b[j]) { losses++; }
```

File: solutions/bronze/usaco-1180.mdx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using Die = array<int, 4>;
+
+/* Create method to calculate whether dice A beats dice B */
+bool beats(const Die &a, const Die &b) {
+	int wins = 0, losses = 0;
+	for (int i = 0; i < 4; i++)
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:32]: braces always

File: solutions/bronze/usaco-1180.mdx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+	for (int i = 0; i < T; i++) {
+		Die A, B;
+		string ans = "no";
+		for (int j = 0; j < 4; j++) cin >> A[j];
+		for (int j = 0; j < 4; j++) cin >> B[j];
+		/* Iterate through all 10,000 possibilities and check if there is a
+		 * possible transitive dice */
+		for (int a = 1; a <= 10; a++)
+			for (int b = 1; b <= 10; b++)
+				for (int c = 1; c <= 10; c++)
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:52]: yeah, like here to

File: solutions/bronze/usaco-1180.mdx [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+			if (a[i] > b[j]) ++wins;
+			if (a[i] < b[j]) ++losses;
+		}
+	return wins > losses;
+}
+
+int main() {
+	int T;
+	cin >> T;
+	for (int i = 0; i < T; i++) {
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:43]: rename T to test_num

File: solutions/bronze/usaco-1180.mdx [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+<PySection>
+
+```py
+# Create method to calculate whether dice A beats dice B
+def win(a, b):
+	return sum([x > y for x in a for y in b]) > sum([y > x for x in a for y in b])
+
+
+t = int(input())
+for _ in range(t):
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:138]:
```suggestion
for _ in range(int(input()))
```

File: solutions/bronze/usaco-1180.mdx [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+	}
+}
+```
+
+</JavaSection>
+<PySection>
+
+```py
+# Create method to calculate whether dice A beats dice B
+def win(a, b):
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1180.mdx:133]:
```suggestion
def win(a, b):
	""":return: whether dice A beats dice B."""
```

Comment (SansPapyrus683) [reply solutions/bronze/usaco-1180.mdx:133]: also type annotations

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]
alright we just need another approval

**Comment by SansPapyrus683** [comment]
remember to change the solution json to fix this build issue
check the contributing page on the guide for what i'm talking about

File: content/extraProblems.json [hunk 12]
```diff
@@ -1164,13 +1164,13 @@
...
+      "name": "Non-Transitive Dice",
+      "url": "http://www.usaco.org/index.php?page=viewproblem2&cpid=1180",
       "source": "Bronze",
-      "difficulty": "Easy",
+      "difficulty": "Normal",
       "isStarred": false,
-      "tags": ["Greedy"],
+      "tags": ["Complete Search"],
       "solutionMetadata": {
         "kind": "internal"
```
Comment (SansPapyrus683) [inline content/extraProblems.json:1175]: why did you completely replace a problem with another

Comment (NimeeshS) [reply content/extraProblems.json:1175]: i thought i just copy and pasted a problem, and then edited the copy to match with the problem i did.

Comment (SansPapyrus683) [reply content/extraProblems.json:1175]: according to the git history, it seems you deleted a problem

Comment (SansPapyrus683) [reply content/extraProblems.json:1175]: bump

Comment (ryanchou-dev) [reply content/extraProblems.json:1175]: 😐 ill fix this pr myself in a few days if they don't respond

Comment (SansPapyrus683) [reply content/extraProblems.json:1175]: just do it now

**Pull request merged**